test/common/config/delta_subscription_impl_test.cc

```diff
@@ -0,0 +1,81 @@
+#include <memory>
+
+#include "common/config/delta_subscription_impl.h"
+#include "common/config/utility.h"
+#include "common/stats/isolated_store_impl.h"
+
+#include "test/mocks/common.h"
+#include "test/mocks/config/mocks.h"
+#include "test/mocks/event/mocks.h"
+#include "test/mocks/grpc/mocks.h"
+#include "test/mocks/local_info/mocks.h"
+#include "test/mocks/runtime/mocks.h"
+#include "test/test_common/logging.h"
+#include "test/test_common/test_time.h"
+#include "test/test_common/utility.h"
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+using testing::NiceMock;
+
+namespace Envoy {
+namespace Config {
+namespace {
+
+TEST(DeltaSubscriptionImplTest, ResourceGoneLeadsToBlankInitialVersion) {
```

Comment:
Nit: can you add a `//` 1 or 2 liner comment here explaining in natural language roughly what is being validated.

Reply:
Done. Also, in revisiting it, I realized I wanted to have the test explore a bit more. Then that testing uncovered another sort-of bug! (Including {"name", ""} entries in the initial_resource_versions sent to the server). PTAL.

Reply:
Actually.... please don't take a look yet! I found a further little change to make, and the new testing is going to be more involved than what there currently is.

Reply:
Ok, *now* I'm pretty confident in the code and the test. PTAL!